Alphakill / Flip control sample issue

It is being broken since https://github.com/DHrpcs3/rpcs3/commit/f8e8e2f624199adf5e994f1b20deab9b0ee05923

-Alphakill 
[E : RSXThread]: Failed to compile shader: 0(16) : error C0000: syntax error, unexpected '.', expecting "::" at token "."

![2](https://cloud.githubusercontent.com/assets/3000282/3282034/86d57a4a-f4d2-11e3-809f-da552d9d8b95.jpg)

-Flip control
![1](https://cloud.githubusercontent.com/assets/3000282/3282031/7190e2c8-f4d2-11e3-9cc7-ec60a3788528.jpg)

Last working build 

-Alphakill
![3](https://cloud.githubusercontent.com/assets/3000282/3282043/eacbaefc-f4d2-11e3-93c0-4e88c32f63c7.jpg)

-Flip control
![4](https://cloud.githubusercontent.com/assets/3000282/3282044/ed37c572-f4d2-11e3-8073-a0c0ac6b9ddc.jpg)

--- a/rpcs3/Emu/GS/GL/GLFragmentProgram.cpp
+++ b/rpcs3/Emu/GS/GL/GLFragmentProgram.cpp
@@ -248,9 +248,9 @@ template<typename T> std::string GLFragmentDecompilerThread::GetSRC(T src)
 std::string GLFragmentDecompilerThread::BuildCode()
 {
 	//main += fmt::Format("\tgl_FragColor = %c0;\n", m_ctrl & 0x40 ? 'r' : 'h');
+	AddCode(m_parr.AddParam(PARAM_OUT, "vec4", "ocol0", 0) + " = " + (m_ctrl & 0x40 ? "r0" : "h0") + ";\n");
 	static const std::pair<std::string, std::string> table[] =
 	{
-		{ "ocol0", m_ctrl & 0x40 ? "r0" : "h0" },
 		{ "ocol1", m_ctrl & 0x40 ? "r2" : "h2" },
 		{ "ocol2", m_ctrl & 0x40 ? "r3" : "h4" },
 		{ "ocol3", m_ctrl & 0x40 ? "r4" : "h6" },
@@ -260,7 +260,7 @@ std::string GLFragmentDecompilerThread::BuildCode()
 	for (int i = 0; i < sizeof(table) / sizeof(*table); ++i)
 	{
 		if (m_parr.HasParam(PARAM_NONE, "vec4", table[i].second))
-			AddCode(m_parr.AddParam(PARAM_OUT, "vec4", table[i].first, i) + " = " + table[i].second + ";");
+			AddCode(m_parr.AddParam(PARAM_OUT, "vec4", table[i].first, i+1) + " = " + table[i].second + ";");
 	}
 
 	if (m_ctrl & 0xe) main += "\tgl_FragDepth = r1.z;\n";

